flake.nix

```diff
@@ -124,7 +114,7 @@
       # Alternatively, we could use nixpkgs.rustfmt,
       # and rebuild with a (stable) fenix toolchain and `asNightly = true`,
       # which would avoid the need to pull another channel altogether.
-      rustfmt-nightly = final.fenix.default.withComponents ["rustfmt"];
+      rustfmt-nightly = final.fenix.stable // final.fenix.stable.rustfmt;
```

Comment:
I believe we need the default here,
default corresponds to "nightly" (go figure ^^) of which we only need the formatter.
We use a few "unstable' i.e. experimental formatting options.